<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="robot" default="spot" />
  <arg name="start_rviz" default="true"/>
  <arg name="rviz_path" default="$(find segment_track_ros)/rviz/$(arg robot).rviz"/>
  <arg name="seg_viz" default="true"/>

  <group ns="$(arg robot)" >

    <remap from="color/image_raw" to="camera/frontleft/image" />
    <remap from="color/camera_info" to="camera/frontleft/camera_info" />
    <remap from="depth/image_raw" to="depth/frontleft/depth_in_visual" />
    <remap from="depth/camera_info" to="depth/frontleft/depth_in_visual/camera_info" />
    <remap from="odom" to="odometry_corrected" />
    
    <node respawn="false" name="fastsam_node" pkg="segment_track_ros" type="fastsam_node.py" output="screen">
        <rosparam command="load" file="$(find segment_track_ros)/cfg/default_fastsam.yaml" />
        <param name="cam_frame_id" value="frontleft_fisheye" />
    </node>

    <node respawn="false" name="segment_track_node" pkg="segment_track_ros" type="segment_track_node.py" output="screen">
        <param name="visualize" value="$(arg seg_viz)" />
        <param name="cam_frame_id" value="frontleft_fisheye" />
        <param name="map_frame_id" value="vision" />
    </node>
  
  </group>

  <node name="segment_track_rviz" pkg="rviz" type="rviz" output="log" if="$(arg start_rviz)" args="-d $(arg rviz_path)"/>

</launch>
